File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        n=len(strs)
        anagrams={}
        for word in strs:
            item=str(sorted(word))
            if item in anagrams:
                anagrams[item].append(word)
            else:
                anagrams[item]=[word]
        result=[]
        for item in anagrams.values():
            result.append(item)
        return result

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> map = new HashMap<>();
        for(int i=0; i<strs.length; i++){
            char[] characters = strs[i].toCharArray();
            Arrays.sort(characters);
            String sortedString = new String(characters);
            List<String> item = map.get(sortedString);
            if(item!=null){
                item.add(strs[i]);
                map.replace(new String(characters), item);
            }else{
                List<String> newItem = new ArrayList<>();
                newItem.add(strs[i]);
                map.put(String.valueOf(characters), newItem);
            }
        }
        List<List<String>> result = new ArrayList<>();
        for(Map.Entry<String, List<String>> entry : map.entrySet()){
            result.add(entry.getValue());
        }
        return result;
    }
}